src/modules/commander/Commander.cpp

```diff
@@ -1640,8 +1659,8 @@ Commander::run()
 					status_changed = true;
 				}
 
-				if (armed.soft_stop != !status.is_rotary_wing) {
-					armed.soft_stop = !status.is_rotary_wing;
+				if (armed.soft_stop != (status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING)) {
```

Comment:
Wow this check is already so ugly before your change. It makes sense to make such checks in multiple stages to descibe, e.g.:
```
const bool is_not_rotary_wing = (status.vehicle_type != vehicle_status_s::VEHICLE_TYPE_ROTARY_WING);
if (armed.soft_stop != is_not_rotary_wing) {
```

Does that make sense? It's still ugly enough I guess.

Reply:
This makes sense. I changed it as you suggested